### ISSUE TITLE
@unocss-devtools-update not found, using incorrect host

### ISSUE THREAD

**Issue by mortona42** [unocss/unocss#3043]
### UnoCSS version

0.51.13

### Describe the bug

I have a Drupal site with a theme that uses vite and unocss. The virtual:unocss-devtools is working fine when I build. I can edit classes in the browser and uno loads the styles. When I'm in dev mode, it throws an error:
(https://devsite.lndo.site/[@unocss-devtools-update](https://devsite.lndo.site/@unocss-devtools-update) [HTTP/2 404 Not Found 315ms]

Drupal and vite are running inside a docker container. The url for Drupal is https://devsite.lndo.site, and vite dev is at https://node.devsite.lndo.site. The devtools update url should be node.devsite.lndo.site/@unocss-devtools-update.

Should I do something with optimize deps or an alias?

### Reproduction

Using this Drupal module: https://www.drupal.org/project/vite

My theme has main.js and styles.css files in /src. I'm using /src/main.js as the rollupOptions input, and it generates manifest.json with /src/main.js mapped to assets/main-xxx.js.

vite.config.js
```
import UnoCSS from 'unocss/vite'
import { defineConfig } from 'vite'
import liveReload from 'vite-plugin-live-reload'

export default(({mode}) => {
  return defineConfig({
    plugins: [
      UnoCSS(),
      liveReload(__dirname + '/**/*.(php|theme|twig|module)'),
    ],

    // Handle base path for dev/build.
    base: mode === 'development' ? '/' : '/themes/landmarks_theme/src',

    build: {
      // Clear out dist directory on build.
      emptyOutDir: true,

      // Create manifest to map library src to dist output.
      manifest: true,

      //
      sourcemap: true,

      rollupOptions: {
        input: './src/main.js',
      },
    },
    // CSS is printed inline in the head,
    // enabling devSourcemap lets it be viewed as a file in the inspector.
    css: { devSourcemap: true },

    server: {
      // SSL is handled by the lando proxy.
      https: false,
      // Listen for connections.
      host: true,
      // Set fixed port 3000.
      port: 3000,
      // Configure hmr.
      hmr: {host: 'node.landmarks.lndo.site', protocol: 'wss'}
    },
  })
})
```

uno.config.js

```
import {
  defineConfig, presetAttributify, presetIcons,
  presetTypography, presetUno, presetWebFonts,
  transformerDirectives, transformerVariantGroup, presetWind
} from 'unocss'


export default defineConfig({
  shortcuts: [
    {
      fontHeader: "text-xl color-primary uppercase font-bold",
    }
  ],
  theme: {
    /* Any color defined below can be used with text- */
    colors: {
      green: '#0F0',
      primary: 'var(--color-brand)',
      secondary: 'var(--color-brand-secondary)',
      offwhite: '#ededed',
      dark: '#333',
    },
    breakpoints: {
      'sm': '640px',
      'md': '768px',
      'lg': '1024px',
      'xl': '1280px',
      '2xl': '1536px'
    }
  },
  rules: [
    [/^text-(.*)$/, ([, c], {theme}) => {
      if (theme.colors[c]) {
        return {
          color: theme.colors[c],
        }
      }
    }],
  ],
  presets: [
    presetUno(),
    presetIcons({
      prefix: 'i-',
      extraProperties: {
        display: 'inline-block'
      }
    }),
    presetTypography(),
    presetWind(),
    presetWebFonts({
      fonts: {
        headerFont: ['Oswald'],
        bodyFont: ['Roboto'],
      },
    }),
  ],
})
```

postcss.config.js

```
export default {
  plugins: {
    '@unocss/postcss': {
      content: ['**/*.{html,twig,theme,js,ts,jsx,tsx,vue,svelte,astro}']
    },
  },
}
```

Setting up an example with lando and drupal is going to be difficult if one is not already familiar. I generated a new theme with drush, installed unocss/vite, and adjusted the libraries according to instructions from the vite module. I also had to proxy port 3000 in the container to a url I could access (could have used the vite port).

### System Info

Using lando (docker containers) on linux mint.

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by mortona42** [comment]
I added {origin: 'https://node.demo.lndo.site'} to the vite server config. Now the url for @unocss-devtools-update is correct, but it is throwing a CORS error.

Now my situation is very similar to #2822. In that one, the error suggests adding 'no-cors' to the request: If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

I changed that in @unocss/vite/dist/client.mjs as suggested in the issue and the error went away.

I don't have a proxy defined in vite.config.js. Would that help here?
